shared/lib/multichain.ts

```diff
@@ -59,3 +65,21 @@ export function getCaipNamespaceFromAddress(address: string): CaipNamespace {
   // Defaults to "Ethereum" for all other cases for now.
   return KnownCaipNamespace.Eip155;
 }
+
+export function isCurrentChainCompatibleWithAccount(
+  chainId: string,
+  account: InternalAccount,
+): boolean {
+  if (!chainId) {
+    return false;
+  }
```

Comment:
Both `chainId` and `account` are required for this method but we only add a check for the `chainId`. Should there be one for `account`?

```ts
if (!chainId || !account) {
  return false;
}
```

Reply:
I think that's a valide point yes, but semantically `account` should be defined (looking at the parameter type).

For `chainId` it can still be defined but be "empty" (`''`). So we could check this, but we should not be able to pass an `undefined` here (unless we `ts-ignore` or come from the "JS-world").

Also, right now we are checking it before the calling point like:
```ts
      account &&
      isCurrentChainCompatibleWithAccount(currentChainId, account) &&
```

Reply:
Also, to avoid being ambiguous here, checking `account` here seems "normal" to narrow down the type to an actual `InternalAccount` and not a potential `InternalAccount | undefined | null` (nullish basically).